the-devastating-chickens/BangazonSite
# User Should be able to delete a payment type

# Description

This pull request contains the code that provides functionality for a soft delete of payment types. It allows payment types with no constraints to be deleted. It removes the payment types that are marked as inactive from the index view.

Fixes Issue # [(10)](https://github.com/the-devastating-chickens/BangazonSite/issues/10)

## Type of change

Please delete options that are not relevant.


# Testing Instructions

1. `git fetch --all`
1. `git checkout cm-deletePaymentType`
1. Review the override SaveChangesAsync method in `ApplicationDbContext.cs`
1. Make sure the code is well commented.
1. Delete the database
1. Go to Nuget PackageManager Console and type `Update-Database`
1. Run the application and navigate to payment types on your account details
1. Try to delete a payment type that is currently under constraint. It should return you to the index view which should show the list of payment types without the one you've deleted.
1. Check your database and ensure the payment type is still there, but has the Active property set to false
1. Create a new payment type, and delete that one and make sure the payment type isn't in the database


# Checklist:
